src/Algebra/Module/Morphism/Linear/Properties.agda

```diff
@@ -0,0 +1,124 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of linear maps.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra                   using (CommutativeRing)
+open import Algebra.Module            using (Module)
+open import Level                     using (Level)
+
+module Algebra.Module.Morphism.Linear.Properties
+  {r ℓr m ℓm : Level}
+  {ring      : CommutativeRing r ℓr}
+  (modA modB  : Module ring m ℓm)
+  where
+
+import Algebra.Module.Properties            as ModuleProperties
+import Algebra.Module.Morphism.Structures   as MorphismStructures
+
+open import Axiom.DoubleNegationElimination
+open import Data.Product
+open import Relation.Binary.Reasoning.MultiSetoid
+open import Relation.Nullary          using (¬_)
+open import Relation.Nullary.Negation using (contraposition)
+
+open MorphismStructures.ModuleMorphisms modA modB
+module A = Module modA
+open A using () renaming (Carrierᴹ to A)
+module B = Module modB
+open B using () renaming (Carrierᴹ to B)
+module PA = ModuleProperties modA
+module PB = ModuleProperties modB
+open CommutativeRing ring renaming (Carrier to S)
+
+module _
+  {f : A → B}
+  (isModuleHomomorphism : IsModuleHomomorphism f)
+  where
+
+  open IsModuleHomomorphism isModuleHomomorphism
+
+  NonTrivial : A → Set (r Level.⊔ m Level.⊔ ℓm)
+  NonTrivial x = (∃₂ λ s y → ((s A.*ₗ x A.≈ᴹ y) × (f y B.≉ᴹ B.0ᴹ)))
+  -- NonTrivial x = Σ[ (s , y) ∈ S × A ] ( (s A.*ₗ x A.≈ᴹ y)
+  --                                     × (f y B.≉ᴹ B.0ᴹ)
+  --                                     )
+
+  x≈0⇒fx≈0 : ∀ {x} → x A.≈ᴹ A.0ᴹ → f x B.≈ᴹ B.0ᴹ
+  x≈0⇒fx≈0 {x} x≈0 = begin⟨ B.≈ᴹ-setoid ⟩
+    f x    ≈⟨ ⟦⟧-cong x≈0 ⟩
+    f A.0ᴹ ≈⟨ 0ᴹ-homo ⟩
+    B.0ᴹ   ∎
+
+  fx≉0⇒x≉0 : ∀ {x} → f x B.≉ᴹ B.0ᴹ → x A.≉ᴹ A.0ᴹ
+  fx≉0⇒x≉0 = contraposition x≈0⇒fx≈0
+
+  -- Zero is a unique output of non-trivial (i.e. - ≉ `const 0`) linear map.
+  zero-unique : ∀ {x} → NonTrivial x → x A.≉ᴹ A.0ᴹ → f x B.≉ᴹ B.0ᴹ
```

Comment:
I'm not sure if the current name is very representative of what the lemma actually says. Maybe we should rename this as `x≉0⇒fx≉0`?